datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -1785,4 +1792,20 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_interval_display() {
```

Comment:
Both test cases result in `(...]`, i.e. they don't really cover the if-else clauses within the formatter. I think having the 4 permutations (`(...)`, `[...)`, `[...]`, `(...]`) would be nice.